### ISSUE TITLE
doomtype.h: MinGW strcasecmp redefinition spam

### ISSUE THREAD

**Issue by AlexMax** [chocolate-doom/chocolate-doom#874]
While working on my CMake files, I noticed something curious from the MinGW's compiler output...

```
In file included from C:\Users\Alex\Documents\Workspace\crispy-doom\src\heretic\doomdata.h:24:0,
                 from C:\Users\Alex\Documents\Workspace\crispy-doom\src\heretic\doomdef.h:34,
                 from C:\Users\Alex\Documents\Workspace\crispy-doom\src\heretic\r_data.c:19:
C:/Users/Alex/Documents/Workspace/crispy-doom/src/doomtype.h:35:0: warning: "strcasecmp" redefined
 #define strcasecmp stricmp
 ^
In file included from C:\Users\Alex\Documents\Workspace\crispy-doom\src\heretic\doomdef.h:22:0,
                 from C:\Users\Alex\Documents\Workspace\crispy-doom\src\heretic\r_data.c:19:
C:/TDM-GCC-64/x86_64-w64-mingw32/include/string.h:110:0: note: this is the location of the previous definition
 #define strcasecmp _stricmp
 ^
```

This is the part of `doomtype.h` it is complaining about:

```
#ifdef _WIN32

#include <string.h>
#define strcasecmp stricmp
#define strncasecmp strnicmp

#else

#include <strings.h>

#endif
```

After some investigation, I found that indeed, mingw-w64 does have definitions of `strcasecmp` and `strcasecmp` in its string.h header file, see [here](https://github.com/mirror/mingw-w64/blob/6172d2f87e426462f89785753f1be553bd10fe2f/mingw-w64-headers/crt/string.h#L107).

Distressingly, I am not able to replicate this warning with the official Cygwin + autotools build process.  The compiler is set to warn about redefinition, which it does in regards to `WIN32_LEAN_AND_MEAN`, but not in regards to `strcasecmp`.  If I take doomtype.h and use it in a small test script with MinGW, I get the warning.  I've been trying to figure out why Cygwin + autotools doesn't complain, and I've got nothing.

Despite my inability to recreate the problem with the official build system, the above redefinition still seems incorrect to me, based on what I've seen of the MinGW string.h file.  I switched the check to this:

```
#ifdef _MSC_VER

#include <string.h>
#define strcasecmp _stricmp
#define strncasecmp _strnicmp

#else

#include <strings.h>

#endif
```

The warnings went away in my CMake build.  The Cygwin autotools build still built correctly, and my MSVC project file did as well.

However, I'm a little hesitant to open a pull request for two reasons:

- I'm not sure if "it messes up my custom build tool" is a valid excuse for a pull request.
- I don't want to break somebody's build without knowing if that fix works in other people's MinGW environments.  I tried the change successfully with CMake MinGW Makefiles, Cygwin + autotools, and Visual Studio 2015, but not a cross-compiling MinGW like you guys probably have set up.

**Comment by chungy** [comment]
If this redefinition is only relevant for MSVC, I think changing the `#ifdef` is the appropriate path to use.

**Comment by haleyjd** [comment]
Yes that is a peculiarity of Visual C++ in particular, not of WIN32 as a platform, so that's not the correct symbol to be using there.
